pkg/webhook/webhook.go

```diff
@@ -129,13 +130,40 @@ func deserializePod(ar *v1beta1.AdmissionReview) (corev1.Pod, error) {
 	/* unmarshal Pod from AdmissionReview request */
 	pod := corev1.Pod{}
 	err := json.Unmarshal(ar.Request.Object.Raw, &pod)
-	/* fix for missing "default" namespace */
-	if pod.ObjectMeta.Namespace == "" {
-		pod.ObjectMeta.Namespace = "default"
+	if pod.ObjectMeta.Namespace != "" {
```

Comment:
I think the case that the pod spec does include a Namespace is valid and should be handled normally. We only need to look up the deployment namespace if there is no Namespace at all.